src/test/java/org/benetech/servicenet/service/impl/ImportServiceConfidentialityImplIntTest.java

```diff
@@ -109,142 +172,113 @@ public void shouldNotCreateLocationIfConfidential() {
+        assertEquals(1, locationService.findAll().size());
+        assertEquals(EXISTING_STRING, locationService.findAll().get(0).getName());
         assertNull(updated);
     }
 
     @Test
     @Transactional
     public void shouldNotCreatePhysicalAddressIfConfidential() {
-        Location location = generateExistingLocation();
-        PhysicalAddress address = new PhysicalAddress().address1(NEW_STRING).city(NEW_STRING).stateProvince(NEW_STRING)
-            .location(location);
+        PhysicalAddress address = new PhysicalAddress().address1(NEW_STRING).city(NEW_STRING).stateProvince(NEW_STRING);
         address.setIsConfidential(true);
-        em.flush();
-        em.refresh(location);
-
-        int dbSize = physicalAddressService.findAll().size();
-        importService.createOrUpdatePhysicalAddress(address, location);
+        Location location = generateExistingLocation().physicalAddress(address);
+        locationImportService.createOrUpdateLocation(location, EXISTING_EXTERNAL_ID, PROVIDER);
 
-        assertEquals(dbSize, physicalAddressService.findAll().size());
+        assertEquals(0, physicalAddressService.findAll().size());
     }
 
     @Test
     @Transactional
     public void shouldNotUpdatePhysicalAddressIfConfidential() {
-        Location location = generateExistingLocation();
-        PhysicalAddress existingAddress = new PhysicalAddress().address1(EXISTING_STRING).city(EXISTING_STRING).stateProvince(EXISTING_STRING)
-            .location(location);
-        em.persist(existingAddress);
-        em.flush();
-        em.refresh(location);
-
-        PhysicalAddress newAddress = new PhysicalAddress().address1(NEW_STRING).city(NEW_STRING).stateProvince(NEW_STRING)
-            .location(location);
-        newAddress.setIsConfidential(true);
+        PhysicalAddress existingAddress = new PhysicalAddress().address1(EXISTING_STRING).city(EXISTING_STRING).stateProvince(EXISTING_STRING);
+        Location location = generateExistingLocation().physicalAddress(existingAddress);
+        locationImportService.createOrUpdateLocation(location, EXISTING_EXTERNAL_ID, PROVIDER);
+        assertEquals(1, physicalAddressService.findAll().size());
 
-        int dbSize = locationService.findAll().size();
-        PhysicalAddress updated = importService.createOrUpdatePhysicalAddress(newAddress, location);
+        PhysicalAddress newAddress = new PhysicalAddress().address1(NEW_STRING).city(NEW_STRING).stateProvince(NEW_STRING);
+        newAddress.setIsConfidential(true);
+        location.setPhysicalAddress(newAddress);
+        locationImportService.createOrUpdateLocation(location, EXISTING_EXTERNAL_ID, PROVIDER);
 
-        assertEquals(dbSize, locationService.findAll().size());
-        assertNull(updated);
+        assertEquals(1, physicalAddressService.findAll().size());
+        assertEquals(EXISTING_STRING, physicalAddressService.findAll().get(0).getAddress1());
```

Comment:
In my opinion, this is misleading because the physical address is outdated.

Reply:
same here. I agree, but we probably have to create a separate a task for it, as it's working the same way as before changing the ImportService.